adapters/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -86,6 +89,7 @@ private void removeListener(OrderedRealmCollection<T> data) {
      * @return the number of items.
      */
     @Override
+    @CheckResult
```

Comment:
Googles own documentation also say that this annotation shouldn't be used on every non-void method: http://tools.android.com/tech-docs/support-annotations. I am assuming because actually checking the constraint isn't free either.

IMO all these methods are called `get*` so the room for misunderstanding what they do is very very small.